plugins/module_utils/ome.py

```diff
@@ -191,18 +196,20 @@ def invoke_request(self, method, path, data=None, query_param=None, headers=None
 
     def __enter__(self):
         """Creates sessions by passing it to header"""
-        if self.req_session:
+        if self.req_session and not self.x_auth_token:
             payload = {'UserName': self.username,
                        'Password': self.password,
                        'SessionType': 'API', }
             path = SESSION_RESOURCE_COLLECTION["SESSION"]
-            resp = self.invoke_request('POST', path, data=payload)
+            resp = self.invoke_request(path, 'POST', data=payload)
```

Comment:
place of passed parameter seems not correct, 

In utils file it's like "invoke_request(self, method, path,"   ---> method and then path, but new changes is path and then method.

reconfirm once from your side.

Reply:
Good observation. I don't know how it got changed.